src/main/java/com/imsweb/mph/DefaultHematoDbUtilsProvider.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2016 Information Management Services, Inc.
+ */
+package com.imsweb.mph;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import au.com.bytecode.opencsv.CSVReader;
+
+public class DefaultHematoDbUtilsProvider implements HematoDbUtilsProvider {
+
+    private static Map<String, List<HematoDbDTO>> _SAME_PRIMARY_DTO = new HashMap<>();
+    private static Map<String, List<HematoDbDTO>> _TRANSFORM_TO_DTO = new HashMap<>();
+    private static Map<String, List<HematoDbDTO>> _TRANSFORM_FROM_DTO = new HashMap<>();
+    private static Pattern _MORPHOLOGY = Pattern.compile("^(\\d{4}/\\d)");
+
+    private DefaultHematoDbUtilsProvider() {
+        initializeLookup();
+    }
+
+    public static DefaultHematoDbUtilsProvider getInstance() {
```

Comment:
A getInstance() method shouldn't instanciate a new object each time, or it defeats the purpose of having an instance! Either you want to force a single instance of this class, and you add a private static variable to hold that instance, and that's what the getInstance() returns, or you remove the getInstance() and make the constructor public (this will force the caller to cache the provider themselves). I would lean towards the second option; it's a bit weird to cache a provider internally I think.